json get して画面描画は完了

s-hayashi1104 opened the pull request:
ただ本来であればproduct.jsを使うべきところだと思う
その後の合計金額を算出するところもproduct.js使いそうなので早めのプルリク

product.jsをrequireで呼び出して使おうとするとis not function と怒られる

s-hayashi1104 committed:
json get して画面描画は完了
ただ本来であればproduct.jsを使うべきところだと思う
その後の合計金額を算出するところもproduct.js使いそうなので早めのプルリク

product.jsをrequireで呼び出して使おうとするとis not function と怒られる

s-hayashi1104 committed:
保存もれ

s-hayashi1104 committed:
選択した商品の合計金額を返すように改修
クエリストリングがids以外の場合400を返すように改修

jonatan-alama commented on server/controller.js:
```diff
@@ -1,5 +1,39 @@
+    if (Number.isNaN(num)) {
+      flag = true;
+    }
+    if (products.length < num) {
```
`products` はidごと確認する必要がないです。

jonatan-alama replied:
すみません、idあるかどうか確認してましたね。

idは基本的に順番で 1,2,3,4 増えてきますが、例えば２番削除すれば、今のコードではうまくいかないですね。

product.jsにcheckProductIdとかのメソッド追加してみてください！

jonatan-alama commented on server/controller.js:
```diff
@@ -1,5 +1,39 @@
+} = require('./product');
+
+const getProductsHandler = function(req, res) {
+  const obj = { 'products': getProducts() };
```
基本的に、jsのobjectのkeyに`'`を使わない (jsonには使う)

s-hayashi1104 committed:
object 作成時に'' をつけないように修正
idの値がすべて存在するかチェックするように処理を追加

Mitsuru53 commented:
@s-hayashi1104 さん
今回のPRに関しては、画面描写されるところまで(金額計算は含まない)という認識ですが、
合っていますでしょうか？
jonatanさんのレビュー対応に関してはokだと思います！

s-hayashi1104 commented:
金額計算含みです！すみません。。。

Mitsuru53 commented:
計算するとundefinedって出てしまいますね。。。
一度ご確認くださいまし
<img width="881" alt="2018-10-11 21 36 02" src="https://user-images.githubusercontent.com/23156390/46804803-8aae3680-cd9e-11e8-9175-f735078139ac.png">

s-hayashi1104 committed:
validateの際数値と文字列を比較していたため数値で比較するように修正

Mitsuru53 commented on server/product.js:
```diff
@@ -7,7 +7,26 @@ const getProducts = function() {
+
 const getTotalPrice = function(productIds) {
+  if (!(checkIds(productIds))) {
+    return false;
```
if (!checkIds(productIds))
でいいはず

Mitsuru53 commented on server/controller.js:
```diff
@@ -1,5 +1,37 @@
+    if (Number.isNaN(num)) {
+      flag = true;
+    }
+  });
```
フラグを用意する必要は特にないかなと思います
```
    if (Number.isNaN(num)) {
      return res.status(400).json({});
    }
```

s-hayashi1104 replied:
それforEachの中なんですよ・・・
for文に書き直しました

Mitsuru53 commented on server/controller.js:
```diff
@@ -1,5 +1,37 @@
+  const obj = {
+    productIds: selectIds,
+    total: price,
+  };
```
変数名を
selectIds ではなく productIds、
priceではなく totalとしておけば
下記のように書けてスッキリするはずです
```
  const obj = {
    productIds,
    total,
  };
```

s-hayashi1104 replied:
変数名とキー名一緒にするとこんな返し方できるんですね。。。
参考になります！！

Mitsuru53 commented on server/product.js:
```diff
@@ -7,7 +7,26 @@ const getProducts = function() {
+  const prods = getProducts();
+
+  prods.forEach((element) => {
+    arr.push(element.id);
```
elementを囲む()は要らない
```
prods.forEach(element => {
```

Mitsuru53 reviewed (commented):
動作は正常なことを確認できました
あとはコードのちょっとしたリファクタを行いましょう！

s-hayashi1104 committed:
controller.js
・フラグを削除
・forEachでは中でreturn出来ないため、for文に変更
・400returnを上記の変更に伴い改修
・変数名を変更
product.js
・forEachをワンライナーに変更

s-hayashi1104 committed:
リファクタリング

Mitsuru53 commented:
OKです！
まーじしちゃってください！
